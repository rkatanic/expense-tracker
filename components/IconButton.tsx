interface Props {
  icon: JSX.Element;
  onClick: () => void;
  type?: "submit" | "button";
  variant?: "primary" | "secondary";
}

const IconButton = ({
  icon,
  onClick,
  type = "button",
  variant = "primary",
}: Props) => {
  return (
    <button
      className={`icon-button icon-button-${variant}`}
      type={type}
      onClick={onClick}
    >
      {icon}
    </button>
  );
};
export default IconButton;
