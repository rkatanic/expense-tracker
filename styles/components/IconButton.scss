.icon-button {
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;
    font-weight: 600;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 1rem;
    cursor: pointer;

    &-primary {
        background-color: var(--color-white-one);

        svg>path {
            stroke: var(--color-black-one);
        }

        &:hover {
            background-color: var(--color-white-two);
        }
    }

    &-secondary {
        border: 1px solid var(--color-gray-700);
        background-color: transparent;
        transition: 0.25s;

        svg>path {
            stroke: var(--color-gray-900);
        }

        &:hover {
            border-color: var(--color-white-one);

            svg>path {
                stroke: var(--color-white-one);
            }
        }
    }
}