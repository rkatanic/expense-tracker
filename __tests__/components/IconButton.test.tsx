import { fireEvent, render } from "@testing-library/react";
import IconButton from "../../components/IconButton";

describe("IconButton", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <IconButton icon={<>icon</>} onClick={jest.fn()} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should call onClick", (): void => {
    const mockOnClick = jest.fn();
    const { getByText } = render(
      <IconButton icon={<>icon</>} onClick={mockOnClick} />
    );

    fireEvent.click(getByText("icon"));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
