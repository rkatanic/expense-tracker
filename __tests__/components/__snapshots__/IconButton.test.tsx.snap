// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render 1`] = `
<body>
  <div>
    <button
      class="icon-button icon-button-primary"
      type="button"
    >
      icon
    </button>
  </div>
</body>
`;
